.github/workflows/push.yml

```diff
@@ -45,9 +99,11 @@ jobs:
         working-directory: docs
         run: JEKYLL_ENV=production bundle exec jekyll build
 
-      - name: Deploy
-        uses: JamesIves/github-pages-deploy-action@3.7.1
+      - name: Upload doc & blueprint artifact
+        uses: actions/upload-pages-artifact@v1
         with:
-          SINGLE_COMMIT: true
-          BRANCH: gh-pages # The branch the action should deploy to.
-          FOLDER: docs/_site
+          path: docs/_site
+
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v1
```

Comment:
Github Pages now no longer push to `gh-pages` branch but directly published. This is needed to fix that Mathlib has huge declaration binaries that `gh-pages` will refuse to commit without a proper `git lfs` set up, also this is the preferred method.

To use this in your repo, you need to go to https://github.com/YaelDillies/LeanAPAP/settings/pages to make it look like this:

<img width="800" alt="image" src="https://github.com/YaelDillies/LeanAPAP/assets/64258/f99bf455-fe33-434f-97a3-c0e71aafb520">